mlrun/api/crud/marketplace.py

```diff
@@ -183,25 +194,56 @@ def get_source_catalog(
     def get_item(
         self,
         source: MarketplaceSource,
-        item_name,
-        channel,
-        version=None,
-        tag=None,
-        force_refresh=False,
+        item_name: str,
+        version: Optional[str] = None,
+        tag: Optional[str] = None,
+        force_refresh: bool = False,
     ) -> MarketplaceItem:
-        catalog = self.get_source_catalog(source, channel, version, tag, force_refresh)
-        items = [item for item in catalog.catalog if item.metadata.name == item_name]
-        if not items:
+        """
+        Retrieve item from source. The item is filtered by tag and version.
+
+        :param source:          Marketplace source object
+        :param item_name:       name of the item to retrieve
+        :param version:         version of the item
+        :param tag:             tag of the item
+        :param force_refresh:   if True, the catalog will be loaded from source always,
+                                otherwise will be pulled from db (if loaded before)
+
+        :return: marketplace item object
+
+        :raise if the number of collected items from catalog is not exactly one.
+        """
+        catalog = self.get_source_catalog(source, version, tag, force_refresh)
+        items, num_items = self._get_catalog_items_filtered_by_name(
+            catalog.catalog, item_name
+        )
```

Comment:
```suggestion
        items = self._get_catalog_items_filtered_by_name(
            catalog.catalog, item_name
        )
        num_items = len(items)
```

As mentioned in my other comment...

Reply:
Done